frontend/src/components/worksheets/BundleDetail/MainContent.jsx

```diff
@@ -0,0 +1,149 @@
+// @flow
+import * as React from 'react';
+import Typography from '@material-ui/core/Typography';
+import CopyIcon from '@material-ui/icons/FileCopy';
+import Tooltip from '@material-ui/core/Tooltip';
+import Grid from '@material-ui/core/Grid';
+import { withStyles } from '@material-ui/core';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+
+import { FileBrowserLite } from '../../FileBrowser';
+
+class MainContent extends React.Component<
+		{
+			bundleInfo: {},
+			stdout: string | null,
+			strerr: string | null,
+			fileContents: string | null,
+			classes: {},
+		}
+> {	
+	
+	render() {
+		const {
+			classes, bundleInfo, stdout, stderr, fileContents } = this.props;
+		const bundleState = (bundleInfo.state == 'running' &&
+							bundleInfo.metadata.run_status != 'Running')
+					? bundleInfo.metadata.run_status
+					: bundleInfo.state;
+        const isRunBundle = bundleInfo.bundle_type === 'run';
+		const stateSpecClass = bundleInfo.state === 'failed'
+            ? 'failedState'
+            : (bundleInfo.state === 'ready' ? 'readyState' : 'otherState');
+		
+		return (
+            <div className={ classes.outter }>
+                <div className={ `${ classes.stateBox } ${ classes[stateSpecClass] }`}>
+                    { bundleState }
+                </div>
+    			<Grid container classes={ { container: classes.container } } spacing={16}>
+                    
+                    { isRunBundle &&

```

Comment:
Can we break these up into separate functions so we don't get these unwieldy huge block?

Reply:
Unless the components are reused somewhere else, I believe it's better form to keep it in one place. We shouldn't shy away from large meaty functions (see Ousterhaut's "deep vs. shallow functions" debate).

But I'll add section headers for clarity & documentation.